Assets/Mirror/Runtime/Transport/Transport.cs

```diff
@@ -117,33 +117,18 @@ public virtual bool Available()
         public abstract void ServerStart();
 
         /// <summary>
-        /// Send data to a client
-        /// </summary>
-        /// <param name="connectionId">The id of the client to send the data to</param>
-        /// <param name="channelId">The channel to be used.  Transports can use channels to implement
-        /// other features such as unreliable, encryption, compression, etc...</param>
-        /// <param name="data"></param>
-        /// <returns>true if the data was sent</returns>
-        public abstract bool ServerSend(int connectionId, int channelId, byte[] data);
-
-        /// <summary>
-        /// Send data to multiple clients at once.
-        /// Calls ServerSend by default. Can be overwritten by transports that can send to multiple
-        /// clients at once more efficiently if needed.
+        /// Send data to one or multiple clients. We provide a list, so that transports can make use
+        /// of multicasting, and avoid allocations where possible.
+        ///
+        /// We don't provide a single ServerSend function to reduce complexity. Simply overwrite this
+        /// one in your Transport.
         /// </summary>
         /// <param name="connectionIds">The list of client connection ids to send the data to</param>
         /// <param name="channelId">The channel to be used.  Transports can use channels to implement
         /// other features such as unreliable, encryption, compression, etc...</param>
         /// <param name="data"></param>
         /// <returns>true if the data was sent to all clients</returns>
-        internal virtual bool ServerSend(List<int> connectionIds, int channelId, byte[] data)
-        {
-            // true by default (for empty lists)
-            bool result = true;
-            foreach (int connectionId in connectionIds)
-                result &= ServerSend(connectionId, channelId, data);
-            return result;
-        }
+        public abstract bool ServerSend(List<int> connectionIds, int channelId, ArraySegment<byte> segment);
```

Comment:
perhaps the default implementation should loop through the connections and call ServerSend(connctionid, channel, segment)?

That way, transports that are naturally allocation free don't have to deal with it?